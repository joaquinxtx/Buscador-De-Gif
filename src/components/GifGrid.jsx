import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'



export const GifGrid = ({category}) => {
    const{images,isLoading}= useFetchGifs(category)

    

  return (
    <>
        <h3>{category} </h3>
        {
          isLoading
          ?
          (<div class="lds-heart"><div></div></div>)
          :
          <div className='card-grid'>
          {images.map(({id,title,url})=>(
            <GifGridItem key={id} title={title} url={url} />
          ))}
          </div>
        }
        
        </>
  )
}
