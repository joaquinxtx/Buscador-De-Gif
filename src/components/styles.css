* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(0, 0, 0);
    
  }
  
  body {
    padding: 60px;
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 50%;
    margin-bottom: 30px;
  }
  h1{
    color:white;
    text-shadow: 1px 1px red;
  }
  
  h2 {
    font-size: 1.5rem;
    
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    -webkit-box-shadow: 0px 4px 14px 0px rgba(242,14,14,1);
    -moz-box-shadow: 0px 4px 14px 0px rgba(242,14,14,1);
    box-shadow: 0px 4px 14px 0px rgba(242,14,14,1);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }

  .lds-heart {
    display: inline-block;
    position: relative;
    width: 80px;
    height: 80px;
    transform: rotate(45deg);
    transform-origin: 40px 40px;
  }
  .lds-heart div {
    top: 32px;
    left: 32px;
    position: absolute;
    width: 32px;
    height: 32px;
    background: rgb(248, 3, 3);
    animation: lds-heart 1.2s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
  }
  .lds-heart div:after,
  .lds-heart div:before {
    content: " ";
    position: absolute;
    display: block;
    width: 32px;
    height: 32px;
    background: rgb(243, 6, 6);
  }
  .lds-heart div:before {
    left: -24px;
    border-radius: 50% 0 0 50%;
  }
  .lds-heart div:after {
    top: -24px;
    border-radius: 50% 50% 0 0;
  }
  @keyframes lds-heart {
    0% {
      transform: scale(0.95);
    }
    5% {
      transform: scale(1.1);
    }
    39% {
      transform: scale(0.85);
    }
    45% {
      transform: scale(1);
    }
    60% {
      transform: scale(0.95);
    }
    100% {
      transform: scale(0.9);
    }
  }