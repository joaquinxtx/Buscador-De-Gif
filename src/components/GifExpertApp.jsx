import React from 'react'
import { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'
import './styles.css'


export const GifExpertApp = () => {
    const [categories, setCategories] = useState([''])
    
    const onAddCategory=(newCategory)=>{
        if(categories.includes(newCategory) )return;
        setCategories([newCategory,...categories])
    }
  return (
    <>
        <h1>Buscador de Gif</h1> 

        <AddCategory 
        
        onNewCategory={onAddCategory}
        />

        {categories.map((category) =>
            <GifGrid 
            key={category} 
            category={category} 
            />
        )}  
   
    </>
  )
}
