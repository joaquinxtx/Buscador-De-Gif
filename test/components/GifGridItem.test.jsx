import { GifGridItem } from "../../src/components/GifGridItem";
import{render, screen} from '@testing-library/react'


describe('Pruebas en <GifGridItem/>', () => {
    const title='saitam'
    const url ='https://jopta.com'
    test('debe de hacer match con el snapshot', () => {
      const{container} = render(<GifGridItem title={title} url={url} />)
      expect(container).toMatchSnapshot()
    });


    test('debe mostrat la imagen y el url indicado', () => {
        render(<GifGridItem title={title} url={url} />)
        // screen.debug()
        // expect(screen.getByRole('img').scr).tobe(url)
        // expect(screen.getByRole('img').alt).tobe(title)
        const{src,alt}=screen.getAllByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(alt)
    });
});