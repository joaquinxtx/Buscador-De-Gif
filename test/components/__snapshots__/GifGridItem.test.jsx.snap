// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitam"
      src="https://jopta.com"
    />
    <p>
      saitam
       
    </p>
  </div>
</div>
`;
